src/internal/connection-provider-routing.js

```diff
@@ -501,3 +501,96 @@ export default class RoutingConnectionProvider extends PooledConnectionProvider
     }
   }
 }
+
+/**
+ * Responsible for keeping track of the existing routing tables
+ */
+class RoutingTableRegistry {
+  /**
+   * Constructor
+   * @param {int} routingTablePurgeDelay The routing table purge delay
+   */
+  constructor (routingTablePurgeDelay) {
+    this._tables = new Map()
+    this._routingTablePurgeDelay = routingTablePurgeDelay
+  }
+
+  /**
+   * Put a routing table in the registry
+   *
+   * @param {string} database The database name
+   * @param {RoutingTable} table The routing table
+   * @returns {RoutingTableRegistry} this
+   */
+  register (database, table) {
+    this._tables.set(database, table)
+    return this
+  }
+
+  /**
+   * Apply function in the routing table for an specific database. If the database name is not defined, the function will
+   * be applied for each element
+   *
+   * @param {string} database The database name
+   * @param {object} callbacks The actions
+   * @param {function (RoutingTable)} callbacks.applyWhenExists Call when the db exists or when the database property is not informed
+   * @param {function ()} callbacks.applyWhenDontExists Call when the database doesn't have the routing table registred
+   * @returns {RoutingTableRegistry} this
+   */
+  apply (database, { applyWhenExists, applyWhenDontExists = () => {} } = {}) {
+    if (this._tables.has(database)) {
+      applyWhenExists(this._tables.get(database))
+    } else if (typeof database === 'string' || database === null) {
+      applyWhenDontExists()
+    } else {
+      this._forEach(applyWhenExists)
+    }
+    return this
+  }
+
+  /**
+   * Retrieves a routing table from a given database name
+   * @param {string} database The database name
+   * @param {function()|RoutingTable} defaultSupplier The routing table supplier, if it's not a function or not exists, it will return itself as default value
+   * @returns {RoutingTable} The routing table for the respective database
+   */
+  get (database, defaultSupplier) {
+    if (this._tables.has(database)) {
+      return this._tables.get(database)
+    }
+    return typeof defaultSupplier === 'function'
+      ? defaultSupplier()
+      : defaultSupplier
+  }
+
+  /**
+   * Remove the routing table which is already expired
+   * @returns {RoutingTableRegistry} this
+   */
+  removeExpired () {
```

Comment:
Oh! Here you remove a routing table for a database. Throwing in comment above should probably not be done. Could be a timeout that kicks in after a long time when the routing table has been removed and in that case you would get wrong type of exception.

Reply:
Agree. This behaviour did not change, but throwing the exception in that situation could stop something to work because of a problem which not exists at all.